gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreFactory.java

```diff
@@ -16,18 +16,29 @@
 
 package com.google.gcloud.datastore;
 
+import com.google.gcloud.ServiceFactory;
+import java.io.ObjectStreamException;
 
 /**
  * A base class for Datastore factories.
  */
-public abstract class DatastoreFactory {
+public abstract class DatastoreFactory implements ServiceFactory<Datastore, DatastoreOptions> {
+
+  private static final long serialVersionUID = 5037190305022535983L;
 
   private static final DatastoreFactory INSTANCE = new DatastoreFactory() {
```

Comment:
I would be more at ease if there were a bit more analysis or documentation. For example, do we expect that an instance serialized today can be deserialized next year? (Are we committing ourselves to serial compatibility?) If we're not guaranteeing that, then what guarantees of serial interop *are* we giving?

I also think that, if we think these are important use cases, there should be tests that exercise them. It's easy to have serialization that appears to work in a simple serialize-then-deserialize case but then fails in practice because, for example, the serialized form of an inner class is pulling in a reference to the containing class.

I'll also reiterate that we must not serialize anonymous classes, in case that got lost in the rest of the discussion.